### PR TITLE
More items in Pipe dispenser and more

### PR DESCRIPTION
**Pull request by Laboredih123**
## About The Pull Request

Adds the Dual port vent, the pressure tank, the multi z pipe, and the large versions of vents, scrubbers, and dual port vents to the RPD.
Adds to pipe-item.dmi to not have the dispenser showing a simple pipe
Makes tanks destructible so you don't have to hack away at a tank every time you misplace one
Minor add of RELEASING and SIPHONING to the DPVENT

## Why It's Good For The Game

More variety and stuff
Cleanliness

## Changelog
:cl:
add: More things in the Pipe Dispenser, check em out
code: Added a define for cleanliness in dpvents
add: Added some sprites to pipe-item
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm [hunk 1]
```diff
@@ -267,6 +267,18 @@
 	playsound(loc, 'sound/weapons/bladeslice.ogg', 100, 1)
 
 
+/obj/machinery/atmospherics/components/unary/vent_scrubber/high_volume
+	name = "large scrubber"
+	idle_power_usage = 25
+	active_power_usage = 150
+	volume_rate = 1000
```
Comment (KubeRoot) [inline code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:274]: 5x the volume rate at less than 3x the active power usage? I'm iffy on the idea of a large scrubber in the first place, but if it *is* included, I think it should have much higher power usage to compensate for the added utility.

File: code/game/objects/items/RPD.dm [hunk 2]
```diff
@@ -22,22 +22,28 @@ GLOBAL_LIST_INIT(atmos_pipe_recipes, list(
...
 		new /datum/pipe_info/meter("Meter"),
-		new /datum/pipe_info/pipe("Gas Filter",			/obj/machinery/atmospherics/components/trinary/filter),
-		new /datum/pipe_info/pipe("Gas Mixer",			/obj/machinery/atmospherics/components/trinary/mixer),
-		new /datum/pipe_info/pipe("Pressure Valve",		/obj/machinery/atmospherics/components/binary/pressure_valve),
-		new /datum/pipe_info/pipe("Temperature Gate",	/obj/machinery/atmospherics/components/binary/temperature_gate),
-		new /datum/pipe_info/pipe("Temperature Pump",	/obj/machinery/atmospherics/components/binary/temperature_pump),
+		new /datum/pipe_info/pipe("Dual-port Air Vent",			/obj/machinery/atmospherics/components/binary/dp_vent_pump),
+		new /datum/pipe_info/pipe("Large Dual-port Air Vent",	/obj/machinery/atmospherics/components/binary/dp_vent_pump/high_volume),
+		new /datum/pipe_info/pipe("Pressure Tank",				/obj/machinery/atmospherics/components/unary/tank),
+
```
Comment (KubeRoot) [inline code/game/objects/items/RPD.dm:46]: Random empty line here

File: code/game/objects/items/RPD.dm [hunk 3]
```diff
@@ -22,22 +22,28 @@ GLOBAL_LIST_INIT(atmos_pipe_recipes, list(
...
+		new /datum/pipe_info/pipe("Large Unary Vent",			/obj/machinery/atmospherics/components/unary/vent_pump/high_volume),
+		new /datum/pipe_info/pipe("Passive Vent",				/obj/machinery/atmospherics/components/unary/passive_vent),
+		new /datum/pipe_info/pipe("Manual Valve",				/obj/machinery/atmospherics/components/binary/valve),
+		new /datum/pipe_info/pipe("Digital Valve",				/obj/machinery/atmospherics/components/binary/valve/digital),
 		new /datum/pipe_info/meter("Meter"),
-		new /datum/pipe_info/pipe("Gas Filter",			/obj/machinery/atmospherics/components/trinary/filter),
-		new /datum/pipe_info/pipe("Gas Mixer",			/obj/machinery/atmospherics/components/trinary/mixer),
-		new /datum/pipe_info/pipe("Pressure Valve",		/obj/machinery/atmospherics/components/binary/pressure_valve),
-		new /datum/pipe_info/pipe("Temperature Gate",	/obj/machinery/atmospherics/components/binary/temperature_gate),
-		new /datum/pipe_info/pipe("Temperature Pump",	/obj/machinery/atmospherics/components/binary/temperature_pump),
```
Comment (KubeRoot) [inline code/game/objects/items/RPD.dm:40]: You appear to be removing the pressure valve, temperature gate and temperature pump from the RPD.

**Review by KubeRoot — CHANGES_REQUESTED** [review]
I'll want to test this in practice at some point, might also want to review potential balance implications from adding buildable large things.

I also think the pressure tank should maybe require material, for example make it so the RPD builds a frame that needs to have plasteel applied to finish building it.

File: code/modules/atmospherics/machinery/components/binary_devices/dp_vent_pump.dm [hunk 4]
```diff
@@ -38,6 +43,12 @@
 	//INPUT_MIN: Do not pass input_pressure_min
 	//OUTPUT_MAX: Do not pass output_pressure_max
 
+	pipe_state = "dpvent"
+	construction_type = /obj/item/pipe/directional
+
+/obj/machinery/atmospherics/components/binary/dp_vent_pump/New()
+	..()
+
```
[the remark this answers is not in the document]
Comment (Laboredih123) [reply code/modules/atmospherics/machinery/components/binary_devices/dp_vent_pump.dm]: Removed

File: code/modules/atmospherics/machinery/components/unary_devices/tank.dm [hunk 5]
```diff
@@ -24,6 +26,54 @@
 		name = "[name] ([GLOB.gas_data.names[gas_type]])"
 	setPipingLayer(piping_layer)
 
+/obj/machinery/atmospherics/components/unary/tank/Destroy()
+	var/datum/gas_mixture/air_contents = airs[1]
+	if(air_contents)
+		var/turf/T = loc
+		T.assume_air(air_contents)
+		air_update_turf()
+	..()
```
[the remark this answers is not in the document]
Comment (Laboredih123) [reply code/modules/atmospherics/machinery/components/unary_devices/tank.dm]: Done

File: code/modules/atmospherics/machinery/components/unary_devices/tank.dm [hunk 6]
```diff
@@ -24,6 +26,54 @@
...
+
+/obj/machinery/atmospherics/components/unary/tank/welder_act(mob/living/user, obj/item/I)
+	..()
```
[the remark this answers is not in the document]
Comment (Laboredih123) [reply code/modules/atmospherics/machinery/components/unary_devices/tank.dm]: Removed as it served no purpose here

File: code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm (hunk 1 above)
Comment (Laboredih123) [reply code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:274]: Values are now 6x for the large scrubber

File: code/game/objects/items/RPD.dm (hunk 3 above)
Comment (Laboredih123) [reply code/game/objects/items/RPD.dm]: Fixed

File: code/game/objects/items/RPD.dm (hunk 2 above)
Comment (Laboredih123) [reply code/game/objects/items/RPD.dm]: Fixed

**Comment by Laboredih123** [comment]
Removed the pressure tank from the RPD, plan to make it a buildable machine.

**Comment by Laboredih123** [comment]
its not meant to close...

**Pull request closed without merging**